docs/src/main/sphinx/installation/query-resiliency.rst

```diff
@@ -0,0 +1,120 @@
+such, it is recommended to run a dedicated fault-tolerant cluster for handling
+batch operations, separate from a cluster that is designated for a higher query
+volume.
+
+Catalogs using the following connectors support fault-tolerant execution of read
+and write operations:
+
+* :doc:`/connector/delta-lake`
+* :doc:`/connector/hive`
+* :doc:`/connector/iceberg`
+* :doc:`/connector/mysql`
+* :doc:`/connector/postgresql`
+* :doc:`/connector/sqlserver`
+
+Catalogs using other connectors only support fault-tolerant execution of read
+operations. When fault-tolerant execution is enabled on a cluster, write
+operations fail on any catalogs that do not support fault-tolerant
+execution of those operations.
+
+The exchange manager may send a large amount of data to the exchange storage,
+resulting in high I/O load on that storage. You can configure multiple storage
+locations for use by the exchange manager to help balance the I/O load between
+them.
+
+Configuration
+-------------
+
+The following steps describe how to configure a Trino cluster for
+fault-tolerant execution with an S3-based exchange:
+
+1. Set up an S3 bucket to use as the exchange storage. For this example we are
+   using an AWS S3 bucket, but other storage options are described in the
+   :doc:`reference documentation </admin/fault-tolerant-execution>`
+   as well. You can use multiple S3 buckets for exchange storage.
+
+   For each bucket in AWS, collect the following information:
+
+   * S3 URI location for the bucket, such as ``s3://exchange-spooling-bucket``
+
+   * Region that the bucket is located in, such as ``us-west-1``
+
+   * AWS access and secret keys for the bucket
+
+2. For a :doc:`Kubernetes deployment of Trino </installation/kubernetes>`, add
+   the following exchange manager configuration in the
+   ``server.exchangeManager`` and ``additionalExchangeManagerProperties``
+   sections of the Helm chart, using the gathered S3 bucket information:
+
+   .. code-block:: yaml
+
+       server:
+         exchangeManager:
+           name=filesystem
+           base-directories=s3://exchange-spooling-bucket-1,s3://exchange-spooling-bucket-2
+
+       additionalExchangeManagerProperties:
+         s3.region=us-west-1
+         s3.aws-access-key=example-access-key
+         s3.aws-secret-key=example-secret-key
+
+   In non-Kubernetes installations, the same properties must be defined in an
+   ``exchange-manager.properties`` configuration file on the coordinator and
+   all worker nodes.
+
+3. Add the following configuration for fault-tolerant execution in the
+   ``additionalConfigProperties:`` section of the Helm chart:
+
+   .. code-block:: yaml
+
+       additionalConfigProperties:
+         retry-policy=TASK
+         exchange.compression-enabled=true
+         query.low-memory-killer.delay=0s
+
+   In non-Kubernetes installations, the same properties must be defined in the
+   ``config.properties`` file on the coordinator and all worker nodes.
```

Comment:
should you merge 3. with 2. to have single section about k8s deployemnt and single section about non-k8s deployment?

Reply:
I think it makes sense to keep in separate steps, because they're broken up by which config file needs to be edited in the non-k8s deployment - `exchange-manager.properties` and `config.properties`, respectively. That may be confusing if condensed into a single step.